internal/cmd/beta/sqlserverflex/user/create/create_test.go

```diff
@@ -0,0 +1,246 @@
+		},
+		InstanceId: testInstanceId,
+		Username:   utils.Ptr("johndoe"),
+		Database:   utils.Ptr("default"),
+		Roles:      utils.Ptr([]string{"read"}),
+	}
+	for _, mod := range mods {
+		mod(model)
+	}
+	return model
+}
+
+func fixtureRequest(mods ...func(request *sqlserverflex.ApiCreateUserRequest)) sqlserverflex.ApiCreateUserRequest {
+	request := testClient.CreateUser(testCtx, testProjectId, testInstanceId)
+	request = request.CreateUserPayload(sqlserverflex.CreateUserPayload{
+		Username: utils.Ptr("johndoe"),
+		Database: utils.Ptr("default"),
+		Roles:    utils.Ptr([]sqlserverflex.Role{"read"}),
+	})
+
+	for _, mod := range mods {
+		mod(&request)
+	}
+	return request
+}
+
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description   string
+		flagValues    map[string]string
+		isValid       bool
+		expectedModel *inputModel
+	}{
+
+		{
+			description:   "base",
+			flagValues:    fixtureFlagValues(),
+			isValid:       true,
+			expectedModel: fixtureInputModel(),
+		},
+		{
+			description: "no username specified",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, usernameFlag)
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.Username = nil
+			}),
+		},
+		{
+			description: "no values",
+			flagValues:  map[string]string{},
+			isValid:     false,
+		},
+		{
+			description: "project id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, projectIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "instance id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, instanceIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "instance id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[instanceIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "database missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, databaseFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "roles missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, roleFlag)
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.Roles = &rolesDefault
+			}),
+		},
+		{
+			description: "invalid role",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[roleFlag] = "invalid-role"
```

Comment:
Why is this invalid? I must be missing something, but isn't this the same as the example in the command?

Reply:
Was a copy paste leftover, for MongoDB roles is an enum